File: doc/listings/mpiinit.f90
program sim
      include 'mpif.h'
      integer :: ierr,id
      integer :: comm
      integer :: is_client
      call MPI_INIT(ierr)
      comm = MPI_COMM_WORLD
      call df_mpi_init_and_start("config.xml",comm,is_client)
      if(is_client.GT.0) then
         call df_mpi_get_client_comm(comm)
         call sim_main_loop(comm)
         call df_kill_server(ierr)
         call df_finalize(ierr)
      endif
      call MPI_FINALIZE(ierr)
end program sim

subroutine sim_main_loop(comm)
       integer :: comm
       integer :: i, ierr
       do i = 1, 100
         ! do something
         call df_end_iteration(i,ierr)
       end do
end subroutine

File: doc/listings/mpiinit.c
#include "include/Damaris.h"

void sim_main_loop(MPI_Comm c)
{
	int i;
	for(i=0;i<100;i++) {
		// do something
		DC_end_iteration();
	}
}
 
int main(int argc, char** argv)
{
	MPI_Init(&argc,&argv);
	if(DC_mpi_init_and_start("config.xml",MPI_COMM_WORLD)) {
		MPI_Comm comm = DC_mpi_get_client_comm();
		sim_main_loop(comm);
		DC_kill_server();
		DC_finalize();
	}
	MPI_Finalize();
	return 0;
}
